lms/djangoapps/courseware/views.py

```diff
@@ -1010,6 +1010,12 @@ def _credit_course_requirements(course_key, student):
     if not (settings.FEATURES.get("ENABLE_CREDIT_ELIGIBILITY", False) and is_credit_course(course_key)):
         return None
 
+    # If student is enrolled as a honor mode, short-circuit and return None. This indicates that
+    # credit requirements should NOT be displayed on the progress page.
+    enrollment = CourseEnrollment.get_enrollment(student, course_key)
+    if enrollment.mode == CourseMode.HONOR:
```

Comment:
Why is this only looking at honor mode? Students in courses going forwards may be enrolled in audit mode as well. As far as I can tell, the logic here should be to only display credit eligibility if the student is enrolled in verified mode or credit mode. That is, we should whitelist those modes instead of blacklisting everything else.